scripts/pylib/twister/twisterlib/statuses.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+# Copyright (c) 2024 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+"""
+Status classes to be used instead of str statuses.
+"""
+
+from enum import Enum
+
+
+class TestInstanceStatus(str, Enum):
+    def __str__(self):
+        return str(self.value)
+
+    NONE = None  # to preserve old functionality
+    ERROR = 'error'
+    FAIL = 'failed'
+    FILTER = 'filtered'
+    PASS = 'passed'
+    SKIP = 'skipped'
+
+
+# Possible direct assignments:
+#   * TestSuiteStatus <- TestInstanceStatus
+class TestSuiteStatus(str, Enum):
+    def __str__(self):
+        return str(self.value)
+
+    NONE = None  # to preserve old functionality
+    FILTER = 'filtered'
+    PASS = 'passed'
+    SKIP = 'skipped'
```

Comment:
Suites do not fail?

Reply:
`TestInstance` can have a `FAIL` status and that status can be assigned to `TestSuite` - thus, `TestSuite` can only fail if its `TestInstance` fails.

Reply:
Unified Status classes into one.